webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/admin/deleteListChildNodeComments.scala.txt

```diff
@@ -0,0 +1,34 @@
+@*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ *@
+
+@import org.knora.webapi.IRI
+
+@**
+ * Deletes a list child node comment.
+ *
+ * @param namedGraph     the named graph to update.
+ * @param nodeIri        the IRI of the list child node to update.
+ * @param isRootNode     flag to identify node type.
+ *@
+@(namedGraph: IRI,
+  nodeIri: IRI,
+  isRootNode: Boolean)
+
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX knora-base: <http://www.knora.org/ontology/knora-base#>
+
+DELETE {
+    GRAPH <@namedGraph> {
+        @if(!isRootNode){
+            <@nodeIri> rdfs:comment ?comments .
+        }
+    }
+}
+
```

Comment:
if I understand the query template correctly, this doesn't do anything, if the node is a root node, right?

If so, I think the request should be refused rather than sending a query to the database that doesn't do anything.

Reply:
See https://github.com/dasch-swiss/dsp-api/pull/2064/files#r878142790.